packages/vincent-contracts/src/facets/VincentAppViewFacet.sol

```diff
@@ -130,6 +121,28 @@ contract VincentAppViewFacet is VincentBase {
         app.delegatees = storedApp.delegatees.values();
     }
 
+    /**
+     * @notice Retrieves the delegatedAgentPkpTokenIds with an offset and limit
+     * @param appId ID of the app to retrieve
+     * @param version Version number of the app to retrieve (1-indexed)
+     * @param offset The offset of the first token ID to retrieve
+     * @param limit The maximum number of token IDs to retrieve
+     * @return delegatedAgentPkpTokenIds Array of delegated agent PKP token IDs
+     */
+    function getDelegatedAgentPkpTokenIds(uint256 appId, uint256 version, uint256 offset, uint256 limit) external view onlyRegisteredAppVersion(appId, version) returns (uint256[] memory delegatedAgentPkpTokenIds) {
+        VincentAppStorage.AppVersion storage versionedApp =
+            VincentAppStorage.appStorage().appIdToApp[appId].appVersions[getAppVersionIndex(version)];
+
+        if (limit == 0 || offset + limit > versionedApp.delegatedAgentPkps.length()) {
+            revert InvalidOffsetOrLimit();
+        }
```

Comment:
nit: Without the check: `|| limit > versionedApp.delegatedAgentPkps.length()` the`.at(offset + i)` will revert with out-of-bounds since the index would exceed the length of the array